cogito/putter.go

```diff
@@ -51,22 +51,40 @@ func (putter *ProdPutter) LoadConfiguration(input []byte, args []string) error {
 		"environment", putter.Request.Env,
 		"args", args)
 
+	// If chat send only is requested, input dir may be empty.
+	// Validate sinks configuration if present, then decide.
+	_, err = putter.Sinks()
+	if err != nil {
+		return fmt.Errorf("put: arguments: %w", err)
+	}
+	sinks := putter.Request.Source.Sinks
+	if len(putter.Request.Params.Sinks) > 0 {
+		sinks = putter.Request.Params.Sinks
+	}
+	sinksSet := sets.From(sinks...)
+
+	// If no configured sinks or magic word 'github' is found, input dir is mandatory.
+	required := len(sinks) == 0 || sinksSet.Contains("github")
+
 	// args[0] contains the path to a directory containing all the "put inputs".
-	if len(args) == 0 {
+	if len(args) == 0 && required {
 		return fmt.Errorf("put: arguments: missing input directory")
 	}
-	putter.InputDir = args[0]
-	putter.log.Debug("", "input-directory", putter.InputDir)
-
+	if len(args) > 0 {
+		putter.InputDir = args[0]
+		putter.log.Debug("", "input-directory", putter.InputDir)
+	}
 	buildState := putter.Request.Params.State
 	putter.log.Debug("", "state", buildState)
 
 	return nil
 }
 
 func (putter *ProdPutter) ProcessInputDir() error {
-	// putter.InputDir, corresponding to key "put:inputs:", should contain 1 or 2 dirs.
-	// If it contains one, we support autodiscovery by not requiring to name it, we know
+	// putter.InputDir, corresponding to key "put:inputs:", should contain 0, 1 or 2 dirs.
+	// If it contains zero, aka undefined, Cogito will only send to chat.
```

Comment:
Note that:
- string `putter.InputDir` is _never_ "undefined" or empty, it is guaranteed to be the path to an existing directory.
- Such existing directory can be empty.
- What can be "undefined" is the "put inputs" configuration key.
- If the "put inputs" configuration key is "undefined", it does NOT imply that putter.InputDir has zero directories: this is the whole point: it could have multiple directories, none of them useful for cogito. This is the famous case where we return an error to help avoiding useless streaming of volumes.

Reply:
Fixed!